src/App.tsx

```diff
@@ -1,19 +1,96 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  FC,
+  useContext,
+  useState,
+  useEffect,
+  useReducer,
+  useMemo,
+  useRef,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { SortType } from './types/Filter';
+
+function filtTodos(
+  todos: Todo[],
+  sortType: SortType,
+) {
+  const visibleTodos = [...todos];
+
+  switch (sortType) {
+    case SortType.Active:
+      return visibleTodos.filter(todo => !todo.completed);
+
+    case SortType.Completed:
+      return visibleTodos.filter(todo => todo.completed);
+    default:
+      return visibleTodos;
+  }
+}
```

Comment:
create helper folder , and import its function where you need

Reply:
But i used this function only in one place and just changing filtType inside the code. Can u describe how to do it better? (Probably one example)

Reply:
Inside `components` folder you could create folder `helpers` with file inside for example `filterTodos.helper.ts` and move this code there. And in App.tsx just import and use it.

In a real project we do this way to make components more readable and simple to understand.

Reply:
@dellachaise thanks, for help i'll do this